src/components/pagination/pagination.jsx

```diff
@@ -0,0 +1,165 @@
+import { createContext, useContext, forwardRef } from 'react';
+import { cn } from '@/utilities/functions';
+import { ChevronLeft, ChevronRight } from 'lucide-react';
+import { disabledClassNames } from './component-style';
+import Button from '../button';
+
+const PageContext = createContext();
+const usePageContext = () => useContext( PageContext );
+
+const Pagination = ( {
+	size = 'sm',
+	disabled = false,
+	children,
+	className,
+	...props
+} ) => (
+	<PageContext.Provider value={ { size, disabled } }>
+		<nav
+			role="navigation"
+			aria-label="pagination"
+			className={ cn(
+				'flex w-full justify-center m-0 box-border',
+				className
+			) }
+			{ ...props }
+		>
+			{ children }
+		</nav>
+	</PageContext.Provider>
+);
+Pagination.displayName = 'Pagination';
+
+const PaginationContent = forwardRef( ( { className, ...props }, ref ) => (
+	<ul
+		ref={ ref }
+		className={ cn(
+			'm-0 w-full flex justify-center flex-row items-center gap-1',
```

Comment:
We might need classes that disable the list style as well, this will not work properly in some pages, where the stylings are added at body level.

Reply:
Fixed.